src/actions/mod.rs

```diff
@@ -197,21 +197,81 @@ impl InitActionContext {
     }
 
     fn build<O: Output>(&self, project_path: &Path, priority: BuildPriority, out: O) {
-        struct BuildNotifier<O: Output> {
+
+        lazy_static! {
+            static ref PROGRESS_ID_COUNTER: AtomicUsize = {
+                AtomicUsize::new(0)
+            };
+        }
+
+        fn new_progress_params(id: String, title: &str) -> ProgressParams {
```

Comment:
As mentioned above, I believe this should also be moved together with `PROGRESS_ID_COUNTER`. When we use this helper function we always generate a new ID atomically, so I'd suggest doing that inside the function instead (so we wouldn't need `id: String` parameter) and accept `title: String` if we always take ownership of it